### ISSUE TITLE
Smaller number of GPUs vs performance

### ISSUE THREAD

**Issue by kwea123** [NVIDIA/semantic-segmentation#31]
I only have 4 GPUs to train on, and get lower mean iou than yours (about 1 point). I know that the number of GPU matters because of batch normalization, but I still wonder if the difference is so significant. Have you trained on 4 GPUs and compared with the results trained on 8 GPUs? Or do you know any paper that does this sort of comparison that I can refer to? Thank you.

**Comment by kwea123** [comment]
Wait, I think I need to know: is the miou in your paper evaluated using [sliding window evaluation](scripts/eval_cityscapes_SEResNeXt50.sh) or pooling (what is this?) or just one pass on the original images (which should also be the number in the checkpoint name during training)?

**Comment by bryanyzhu** [comment]
@kwea123 Pooling = one pass + flipping

The mIoU in my paper (for ablation study) is evaluated using one pass, and it is the number in the checkpoint name. The mIoU (for sota comparison) is evaluated using mutli-scale + sliding + flipping. 

Your numbers look good. I have the same numbers.

**Comment by bryanyzhu** [comment]
For your question on **smaller number of GPUs vs performance**

I have trained with 4 GPUs, 6 GPUs and 8 GPUs before. Compared to normal setting (using 8 GPUs), usually 4 GPUs give me about 0.7% mIoU decrease. 6 GPUs give me about 0.3% mIoU decrease. If you have try on less than 4 GPUs, the number starts to drop significantly because the batch size is too small. Hope this is helpful.

**Comment by kwea123** [comment]
Thanks. So I think my 1% decrease is normal. I'll make the comparison between the methods first using 4 GPUs than compare with your results once I have access to 8 GPUs.

**Comment by kwea123** [comment]
Sorry, I use the exact same script to train on both 4 and 8 GPUs (the default script with snapshot mapillary and maxSkip 3), but I got lower mIoU when training on 8 GPUs, and the result is much lower than what you report in the paper. I have around 80.5 for 4 GPU and 80.2 for 8 GPU. Can you tell where I am wrong?

**Comment by bryanyzhu** [comment]
It seems from the plot that, the training IoU of 8 GPU (red) is higher than 4 GPU (orange). 

Did these numbers (80.5 for 4 GPU and 80.2 for 8 GPU) comes from validation?

**Comment by kwea123** [comment]
Yes, this is the training graph for original training set. The 80.5 and 80.2 are for +-3. What I wanted to show with this graph is that I cannot reach your high mIoU: I only get 80.04 but you report 80.85.

The numbers are all taken from the checkpoint name, so from validation yes.

**Comment by kwea123** [comment]
I found the problem. I was using mapillary commercial version to pretrain. After changing to **mapillary research** version, I get high accuracy as your paper.
